src/execution/vm/bytecode_generator.cpp

```diff
@@ -503,10 +503,11 @@ void BytecodeGenerator::VisitReturnStmt(ast::ReturnStmt *node) {
     LocalVar result;
     if (node->Ret()->GetType()->IsSqlValueType()) {
       result = VisitExpressionForSQLValue(node->Ret());
+      BuildAssign(rv, result, node->Ret()->GetType());
```

Comment:
@pmenon Could you check this one? Thanks!

Reply:
Since you already have a destination address to store the result, you can get by with:
```
if (node->Ret() != nullptr) {
  LocalVar rv = GetCurrentFunction()->GetReturnValueLocal();
  TERRIER_ASSERT(rv.GetAddressMode() == LocalVar::AddressMode::Address, "RV expected to be address.");
  // rv is a double pointer to the result. So we want the recursive call to store the
  // result of the computation into *rv. Hence, we set the destination to rv.ValueOf().
  VisitExpressionForRValue(node->Ret(), rv.ValueOf());
}
```